src/E225B0.c

```diff
@@ -1,7 +1,203 @@
+    /* 0x10 */ u8 unk_10;
+} UnknownCommand;
+
+void func_8013AF70(s32, UnknownCommand*, s32, Matrix4f*); // extern
+
+extern void (*D_8010C93C)(void);
+extern struct802B7C78* D_802B7C78_E23228;
+void func_802B735C_E2290C(void);
+
+void func_802B7000_E225B0(void) {
+    Matrix4f matrix1;
+    Matrix4f matrix2;
+    UnknownCommand command;
+
+    Gfx* oldMasterGfxPos;
+    f32 scale;
+    s32 temp;
+    s32 matrixIndex;
+    struct802B7C78* localD_802B7C78_E23228;
+    u16 oldMatrixListPos;
+
+    if ((gPlayerStatus.animFlags & 0x100) != 0) {
+        guScaleF(matrix1, D_802B7C78_E23228->scale, D_802B7C78_E23228->scale,
+            D_802B7C78_E23228->scale);
+        guRotateF(matrix2, -gCameras[gCurrentCameraID].currentYaw, 0.0f, 1.0f, 0.0f);
+        guMtxCatF(matrix1, matrix2, matrix1);
+        guTranslateF(matrix2, D_802B7C78_E23228->x, D_802B7C78_E23228->y, D_802B7C78_E23228->z);
+        guMtxCatF(matrix1, matrix2, matrix2);
+        guMtxF2L(matrix2, (Mtx*) &gDisplayContext->matrixStack[gMatrixListPos]);
+
+        gSPMatrix(gMasterGfxPos++, &gDisplayContext->matrixStack[gMatrixListPos++], 3);
+        gSPDisplayList(gMasterGfxPos++, (u32) &D_802B7C00_E231B0);
+
+        localD_802B7C78_E23228 = D_802B7C78_E23228;
+        if (localD_802B7C78_E23228->unk_18 < 0x2F) {
+            localD_802B7C78_E23228->unk_20 += 1;
+        }
+
+        temp = D_802B7C78_E23228->unk_20;
+        temp = temp - (temp / 12) * 12;
+        switch (temp) {
+        case 0:
+        case 1:
+        case 2:
+        case 3:
+            // spA4 = &D_802B7BA0_E23150;
+            command.unk_04 = &D_802B7BA0_E23150;
+            break;
+        case 4:
+        case 5:
+        case 6:
+        case 7:
+            // spA4 = &D_802B7BC0_E23170;
+            command.unk_04 = &D_802B7BC0_E23170;
+            break;
+        case 8:
+        case 9:
+        case 10:
+        case 11:
+            // spA4 = &D_802B7BE0_E23190;
+            command.unk_04 = &D_802B7BE0_E23190;
+            break;
+        }
+        func_8013AA9C(0, 7, 0xFF, 0xFF, 0xFF, D_802B7C78_E23228->unk_28, 0);
+
+        command.unk_00 = &D_802B7580_E22B30;
+        command.unk_08 = 0x38;
+        command.unk_0A = 0x38;
+        command.unk_0C = -0x1C;
+        command.unk_0E = 0x2E;
+        command.unk_10 = 0xFF;
+
+        func_8013AF70(0, &command, 0, &matrix2);
+        gSPPopMatrix(gMasterGfxPos++, 0);
+    }
+}
+
+
+void func_802B72C0_E22870(void) {
+    mem_clear((s8*)D_802B7C78_E23228, 0x2C);
+
+    D_802B7C78_E23228->x = (f32)gPlayerStatus.position.x;
+    D_802B7C78_E23228->y = (f32)(gPlayerStatus.position.y + (f32)gPlayerStatus.colliderHeight + 8.0f);
+    D_802B7C78_E23228->z = (f32)gPlayerStatus.position.z;
+
+    D_802B7C78_E23228->unk_28 = 0xFF;
+
+    gPlayerStatus.animFlags |= 0x100;
+    D_8010C93C = &func_802B735C_E2290C;
+}
+
+
+typedef struct struct8015A578 {
+    u8 unk_0[0x2];
+    u8 unk_2;
+    u8 unk_3[0x9];
+} struct8015A578;
+extern struct8015A578 D_8015A578;
```

Comment:
We should just put this into common_structs.h at this point since it's in 2 other files as well as this one

Reply:
Refactor for this will be in the next commit.